crates/ruff_python_formatter/src/other/f_string_element.rs

```diff
@@ -0,0 +1,308 @@
+                    text(formatted.print()?.as_code()).fmt(f)?;
+                } else {
+                    expression.format().fmt(f)?;
+                }
+
+                // Conversion comes first, then the format spec.
+                match conversion {
+                    ConversionFlag::Str => text("!s").fmt(f)?,
+                    ConversionFlag::Ascii => text("!a").fmt(f)?,
+                    ConversionFlag::Repr => text("!r").fmt(f)?,
+                    ConversionFlag::None => (),
+                }
+
+                if let Some(format_spec) = format_spec.as_deref() {
+                    let elements =
+                        format_with(|f| {
+                            f.join()
+                                .entries(format_spec.elements.iter().map(|element| {
+                                    FormatFStringElement::new(element, self.context)
+                                }))
+                                .finish()
+                        });
+                    write!(
+                        f,
+                        [
+                            token(":"),
+                            elements,
+                            trailing_comments(trailing_format_spec_comments)
+                        ]
+                    )?;
+                }
+
+                line_break_or_space.fmt(f)
+            });
+
+            let inner = format_with(|f| {
+                let mut buffer = RemoveSoftLinesBuffer::new(f);
+
+                if dangling_open_parentheses_comments.is_empty() {
+                    if self.context.should_remove_soft_line_breaks() {
+                        write!(buffer, [group(&soft_block_indent(&item))])
+                    } else {
+                        write!(f, [group(&soft_block_indent(&item))])
+                    }
+                } else {
+                    if self.context.should_remove_soft_line_breaks() {
+                        write!(
+                            buffer,
+                            [group(&format_args![
+                                dangling_open_parenthesis_comments(
+                                    dangling_open_parentheses_comments
+                                ),
+                                soft_block_indent(&item),
+                            ])]
+                        )
+                    } else {
+                        write!(
+                            f,
+                            [group(&format_args![
+                                dangling_open_parenthesis_comments(
+                                    dangling_open_parentheses_comments
+                                ),
+                                soft_block_indent(&item),
+                            ])]
+                        )
+                    }
+                }
+            });
+
+            let mut f = WithNodeLevel::new(NodeLevel::ParenthesizedExpression, f);
+
+            write!(f, [token("{"), inner, token("}")])
```

Comment:
I think we can reduce the code duplication here by making use of `Option` as you did above with the spacing. 


```suggestion
            let open_parenthesis_comments = (!dangling_item_comments.is_empty()).then_some(
                dangling_open_parenthesis_comments(dangling_open_parentheses_comments),
            );

            write!(f, [token("{")])?;

            {
                let mut f = WithNodeLevel::new(NodeLevel::ParenthesizedExpression, f);
                if self.context.should_remove_soft_line_breaks() {
                    let mut buffer = RemoveSoftLinesBuffer::new(&mut *f);
                    write!(buffer, [open_parenthesis_comments, item])?;
                } else {
                    group(&format_args![
                        open_parenthesis_comments,
                        soft_block_indent(&item)
                    ])
                    .fmt(&mut f)?;
                }
            }

            token("}").fmt(f)
```

Reply:
Thanks, this simplifies a lot!